include/deal.II/matrix_free/mapping_info.templates.h

```diff
@@ -287,19 +311,79 @@ namespace internal
+             update_quadrature_points ?
+           update_quadrature_points :
+           update_default) |
+        update_normal_vectors | update_JxW_values | update_jacobians;
+      this->update_flags_inner_faces    = this->update_flags_boundary_faces;
+      this->update_flags_faces_by_cells = update_flags_faces_by_cells;
+
+      for (unsigned int my_q = 0; my_q < quad.size(); ++my_q)
+        {
+          const unsigned int n_hp_quads = quad[my_q].size();
+          AssertIndexRange(0, n_hp_quads);
+          cell_data[my_q].descriptor.resize(n_hp_quads);
+          for (unsigned int q = 0; q < n_hp_quads; ++q)
+            cell_data[my_q].descriptor[q].initialize(quad[my_q][q],
+                                                     update_default);
+
+          face_data[my_q].descriptor.resize(n_hp_quads);
+          for (unsigned int hpq = 0; hpq < n_hp_quads; ++hpq)
+            face_data[my_q].descriptor[hpq].initialize(
+              quad[my_q][hpq], update_flags_boundary_faces);
+
+          face_data_by_cells[my_q].descriptor.resize(n_hp_quads);
+          for (unsigned int hpq = 0; hpq < n_hp_quads; ++hpq)
+            face_data_by_cells[my_q].descriptor[hpq].initialize(quad[my_q][hpq],
+                                                                update_default);
+        }
+
       // Could call these functions in parallel, but not useful because the
       // work inside is nicely split up already
-      initialize_cells(
-        tria, cells, active_fe_index, mapping, quad, update_flags_cells);
-      initialize_faces(tria,
-                       cells,
-                       face_info.faces,
-                       mapping,
-                       quad,
-                       update_flags_boundary_faces,
-                       update_flags_inner_faces);
-      initialize_faces_by_cells(
-        tria, cells, mapping, quad, update_flags_faces_by_cells);
+      initialize_cells(tria, cells, active_fe_index, mapping);
+      initialize_faces(tria, cells, face_info.faces, mapping);
+      initialize_faces_by_cells(tria, cells, mapping);
+    }
+
+
+
+    template <int dim, typename Number, typename VectorizedArrayType>
+    void
+    MappingInfo<dim, Number, VectorizedArrayType>::update_mapping(
+      const dealii::Triangulation<dim> &                        tria,
+      const std::vector<std::pair<unsigned int, unsigned int>> &cells,
+      const FaceInfo<VectorizedArrayType::n_array_elements> &   face_info,
+      const std::vector<unsigned int> &                         active_fe_index,
+      const Mapping<dim> &                                      mapping)
+    {
+      AssertDimension(cells.size() / VectorizedArrayType::n_array_elements,
+                      cell_type.size());
+
+      for (auto &data : cell_data)
+        data.clear_data_fields();
+      for (auto &data : face_data)
+        data.clear_data_fields();
+      for (auto &data : face_data_by_cells)
+        data.clear_data_fields();
```

Comment:
I this necessary?

Reply:
In general yes I would say - think of a deforming mesh case where we start with a Cartesian mesh but due to deformation end up with a general type. This will necessitate different array sizes, so I thought it is best to refresh the arrays. I make some assumptions about NUMA in there (even though the threaded case is not well-tuned) that would be affected.

Reply:
> we start with a Cartesian mesh but due to deformation end up with a general type

That is true. I did not think of this case!